core/primitives/src/shard_layout.rs

```diff
@@ -619,34 +619,46 @@ impl ShardLayout {
     /// Only calls this function for shard layout that has parent shard layouts
     /// Returns error if `shard_id` is an invalid shard id in the current layout
     /// Panics if `self` has no parent shard layout
-    pub fn get_parent_shard_id(&self, shard_id: ShardId) -> Result<ShardId, ShardLayoutError> {
+    pub fn try_get_parent_shard_id(
+        &self,
+        shard_id: ShardId,
+    ) -> Result<Option<ShardId>, ShardLayoutError> {
         if !self.shard_ids().any(|id| id == shard_id) {
             return Err(ShardLayoutError::InvalidShardIdError { shard_id });
         }
         let parent_shard_id = match self {
-            Self::V0(_) => panic!("shard layout has no parent shard"),
+            Self::V0(_) => None,
             Self::V1(v1) => match &v1.to_parent_shard_map {
                 // we can safely unwrap here because the construction of to_parent_shard_map guarantees
                 // that every shard has a parent shard
                 Some(to_parent_shard_map) => {
                     let shard_index = self.get_shard_index(shard_id)?;
-                    *to_parent_shard_map.get(shard_index).unwrap()
+                    Some(*to_parent_shard_map.get(shard_index).unwrap())
                 }
-                None => panic!("shard_layout has no parent shard"),
+                None => None,
             },
             Self::V2(v2) => match &v2.shards_parent_map {
                 Some(to_parent_shard_map) => {
                     let parent_shard_id = to_parent_shard_map.get(&shard_id);
                     let parent_shard_id = parent_shard_id
                         .ok_or(ShardLayoutError::InvalidShardIdError { shard_id })?;
-                    *parent_shard_id
+                    Some(*parent_shard_id)
                 }
-                None => panic!("shard_layout has no parent shard"),
+                None => None,
             },
         };
         Ok(parent_shard_id)
     }
 
+    /// Return the parent shard id for a given shard in the shard layout.
+    /// Returns error if `shard_id` is an invalid shard id in the current
+    /// layout. Returns None if the shard layout has no parent shard layout.
+    pub fn get_parent_shard_id(&self, shard_id: ShardId) -> Result<ShardId, ShardLayoutError> {
```

Comment:
Oh, I guess comment section of `try_get_parent_shard_id` and `get_parent_shard_id` got switched between each other